### PR TITLE
Add domain tags to Decision/Activity timeout metrics

### PR DESCRIPTION
**Pull request by longquanzheng**

### CODE REVIEW

File: service/history/timerQueueActiveProcessor.go [hunk 1]
```diff
@@ -430,10 +430,17 @@ ExpireActivityTimers:
 			}
 		}
 
+		domainEntry, err := t.shard.GetDomainCache().GetDomainByID(msBuilder.GetExecutionInfo().DomainID)
+		if err != nil {
+			return &workflow.InternalServiceError{Message: "Unable to get domain from cache by domainID."}
```
Comment (yux0) [inline service/history/timerQueueActiveProcessor.go]: nit: first letter lower case for the error message.

File: service/history/timerQueueActiveProcessor.go [hunk 2]
```diff
@@ -520,17 +533,26 @@ func (t *timerQueueActiveProcessorImpl) processDecisionTimeout(
 		return nil
 	}
 
+	domainEntry, err := t.shard.GetDomainCache().GetDomainByID(msBuilder.GetExecutionInfo().DomainID)
+	if err != nil {
+		return &workflow.InternalServiceError{Message: "Unable to get domain from cache by domainID."}
```
Comment (yux0) [inline service/history/timerQueueActiveProcessor.go]: same

File: service/history/timerQueueActiveProcessor.go (hunk 1 above)
Comment (longquanzheng) [reply service/history/timerQueueActiveProcessor.go]: OK

File: service/history/timerQueueActiveProcessor.go [hunk 3]
```diff
@@ -430,10 +430,17 @@ ExpireActivityTimers:
...
+		domainEntry, err := t.shard.GetDomainCache().GetDomainByID(msBuilder.GetExecutionInfo().DomainID)
+		if err != nil {
+			return &workflow.InternalServiceError{Message: "unable to get domain from cache by domainID."}
+		}
+
 		switch timeoutType {
 		case workflow.TimeoutTypeScheduleToClose:
 			{
-				t.metricsClient.IncCounter(metrics.TimerActiveTaskActivityTimeoutScope, metrics.ScheduleToCloseTimeoutCounter)
+				t.metricsClient.Scope(metrics.TimerActiveTaskActivityTimeoutScope).
```
Comment (vancexu) [inline service/history/timerQueueActiveProcessor.go]: Should we tag scope out so that scope and `.Scope` and `.Tag` will be reused?

Comment (longquanzheng) [reply service/history/timerQueueActiveProcessor.go]: It can't be reused if you look into the code. The scopes are different for each type.

Comment (vancexu) [reply service/history/timerQueueActiveProcessor.go]: Probably I didn't explain clear. I saw several duplicates
```
t.metricsClient.Scope(metrics.TimerActiveTaskActivityTimeoutScope).
					Tagged(metrics.DomainTag(domainEntry.GetInfo().Name)).
```
and I think we could reduce that. https://github.com/uber/cadence/pull/2386

**Pull request merged**